src/BalancedBagging.jl

```diff
@@ -0,0 +1,232 @@
+
+"""
+Return a dictionary `result` mapping each unique value in a given abstract vector `y`
+    to the vector of indices where that value occurs.
+"""
+function group_inds(y::AbstractVector{T}) where {T}
+    result = LittleDict{T,AbstractVector{Int}}()
+    freeze(result)
+    for (i, v) in enumerate(y)
+        # Make a new entry in the dict if it doesn't exist
+        if !haskey(result, v)
+            result[v] = []
+        end
+        # It exists, so push the index belonging to the class
+        push!(result[v], i)
+    end
+    return result
```

Comment:
If you no longer need to mutate, then you can use `freeze(result)` for faster lookup.

```suggestion
    return freeze(result)
```

Reply:
Understood.